python/ray/_raylet.pyx

```diff
@@ -4947,3 +4950,43 @@ def get_session_key_from_storage(host, port, username, password, use_ssl, config
     else:
         logger.info("Could not retrieve session key from storage.")
         return None
+
+
+# Authentication mode enum exposed to Python
```

Comment:
let's put these in a different file instead of piling into `_raylet`

example: https://github.com/ray-project/ray/blob/77a96d9df09fe4a0731c04448b79b36c129840d3/python/ray/includes/function_descriptor.pxi